src/modals.js

```diff
@@ -305,7 +305,7 @@ function CreateNewProject() {
     // event handler from storing the old project code into
     // the browser storage.
     // ------------------------------------------------------
```

Comment:
Turns out this was the culprit.  `{}` means that `!projectData` is still false - so it was breaking the newProject page reload.

Reply:
So that we're on the same page with projectData;
* It can only exist in one of two states
    * It is null
    * It contains the original copy of the currently loaded project.
* It references an object and never a string.

The projectData object will eventually become an instance of the Project class.

Reply:
I agree, not sure not this ended up here.